frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx

```diff
@@ -197,12 +199,12 @@ const TestQueryBuilder = (
 
 const TestHome = () => <div />;
 
-function isSavedCard(card: Card | UnsavedCard): card is Card {
-  return "id" in card;
+function isSavedCard(card: Card | UnsavedCard | null): card is Card {
+  return card !== null && "id" in card;
 }
 
 interface SetupOpts {
-  card?: Card | UnsavedCard;
+  card?: Card | UnsavedCard | null;
```

Comment:
There is no `card` when we're creating a new question, so `null` is now allowed to indicate that.

Reply:
why card can't be `undefined` in this case?

Reply:
There's a [default value for this prop](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx#L211). We could refactor it I guess. But even then, [I'd say `null` should represent the lack of card](https://metaboat.slack.com/archives/C505ZNNH4/p1695226612671479?thread_ts=1695225341.069819&cid=C505ZNNH4).

Reply:
+1 to refactor the default value. We shouldn't use `null` and `undefined` for different things as much as possible.

Reply:
I removed the default value: e958c742c93c66f887c721ad0e35777fc94201b1. Now only `null` is allowed.